CMakeLists.txt

```diff
@@ -56,6 +56,14 @@ set( CMAKE_CXX_FLAGS "${flags_str}${CMAKE_CXX_FLAGS}")
 
 set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O3 -L${ROCM_PATH}/lib")
 include_directories(${ROCM_PATH}/include)
+find_library(IBVERBS_LIBRARY ibverbs)
+if (IBVERBS_LIBRARY)
+    message(STATUS "Found ibverbs: ${IBVERBS_LIBRARY}")
+    link_libraries(ibverbs)    
+else()
+    add_definitions(-DNO_IBV_EXEC)
```

Comment:
I agree this is counterintuitive. My only reason for doing so is that I have noted that many folks, including me recently, use VS code + Intellisense. Since Intellisense doesn't see it as defined, huge portions of the code appear grayed out. Alternatively, we can track a .vscode/settings.json file where we define the environment.

Reply:
I'd recommend writing the code without the negative and individuals can define things in their own environment.

Reply:
> If you wanted to satisfy both, you could use IBV_EXEC_ENABLED throughout the code, then at top of header use #ifndef NO_IBV_EXEC #define IBV_EXEC_ENABLED #endif. But I think either is probably fine.

I like this idea @corey-derochie-amd. I think when we cover Slingshot NICs, opting out with a disable macro will make more sense than disabling NIC exec by default